javalib/src/main/scala/java/math/BigInteger.scala

```diff
@@ -0,0 +1,1449 @@
+/*
+ *  Ported by Alistair Johnson from  https://github.com/gwtproject/gwt/blob/master/user/super/com/google/gwt/emul/java/math/BigInteger.java
+ *  Original license copied below:
+ */
+
+/*
+ * Copyright 2009 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * INCLUDES MODIFICATIONS BY RICHARD ZSCHECH AS WELL AS GOOGLE.
+ */
+
+package java.math
+
+import java.io.Serializable
+import java.util.Random
+import scala.annotation.tailrec
+
+/**
+ * Companion object for {@code BigInteger}
+ */
+object BigInteger {
+
+  /**
+   * The {@code BigInteger} constant 1.
+   */
+  val ONE = new BigInteger(1, 1)
+
+  /**
+   * The {@code BigInteger} constant 10.
+   */
+  val TEN = new BigInteger(1, 10)
+
+  /**
+   * The {@code BigInteger} constant 0.
+   */
+  val ZERO = new BigInteger(0, 0)
+
+  /**
+   * The {@code BigInteger} constant 0 used for comparison.
+   */
+  private[math] val EQUALS = 0
```

Comment:
`final val`. Same for `GREATER`, `LESS`, and `POW32` (because they're literals, they can then be constant-folded by scalac)